text_reading/src/main/scala/org/clulab/grounding/Grounder.scala

```diff
@@ -0,0 +1,222 @@
+  }
+
+  /* Grounding a sequence of mentions and return a pretty-printable json string*/
+  def mentionsToGroundingsJson(mentions: Seq[Mention], k: Int): String = {
+    val seqOfGroundings = groundDefinitions(mentions, k)
+    write(seqOfGroundings, indent = 4)
+  }
+
+  /* grounding one mention; return a map from the name if the variable from the mention to its svo grounding */
+  def groundOneMentionWithSparql(mention: Mention, k: Int): Map[String, Seq[sparqlResult]] = {
+
+    val terms = getTerms(mention) //get terms gets nouns, verbs, and adjs, and also returns reasonable collocations (multi-word combinations), e.g., syntactic head of the mention + (>compound | >amod)
+    if (terms.nonEmpty) {
+      val resultsFromAllTerms = new ArrayBuffer[sparqlResult]()
+      for (word <- terms.get) {
+        val result = runSparqlQuery(word, "/home/alexeeva/Repos/automates/text_reading/sparql") //todo: pass through configs
+        if (result.nonEmpty) {
+          //each line in the result is a separate entry returned by the query:
+          val resultLines = result.split(("\n"))
+          //represent each returned entry/line as a sparqlResult and sort those by edit distance score (lower score is better)
+          val sparqlResults = resultLines.map(rl => new sparqlResult(rl.split("\t")(0), rl.split("\t")(1), rl.split("\t")(2), Some(editDistance(rl.split("\t")(1), word)), "SVO")).sortBy(sr => sr.score)
+          for (sr <- sparqlResults) resultsFromAllTerms += sr
+        }
+       resultsFromAllTerms.toArray
+      }
+
+      //RANKING THE RESULTS (//todo: there has to be a more efficient way)
+      //getting all the results with same (minimal) score
+      val onlyMinScoreResults = resultsFromAllTerms.filter(res => res.score == resultsFromAllTerms.map(r => r.score).min).toArray.distinct
+
+      //the results where search term contains "_" or "-" should be ranked higher since those are multi-word instead of separate words
+      val (multiWord, singleWord) = onlyMinScoreResults.partition(r => (r.searchTerm.contains("_") || r.searchTerm.contains("-")))
+
+      //this is the best results based on score and whether or not they are collocations
+      val bestResults = if (multiWord.nonEmpty) multiWord else singleWord
+
+      //return the best result first, then only the min score ones, and then all the rest; some may overlap thus distinct
+
+      val allResults = (bestResults ++ onlyMinScoreResults ++ resultsFromAllTerms).distinct
+      Map(mention.arguments("variable").head.text -> getTopK(allResults, k))
+    } else Map(mention.arguments("variable").head.text -> Array(new sparqlResult("None", "None", "None", None)))
+  }
+
+  //ground a string (for API)
+  def groundString(text:String): String = {
+    val terms = getTerms(text)
+    val resultsFromAllTerms = new ArrayBuffer[sparqlResult]()
+    for (word <- terms) {
+      val result = runSparqlQuery(word, "/home/alexeeva/Repos/automates/text_reading/sparql") //todo: pass from configs
```

Comment:
should we change this for this PR?

Reply:
changed this to passing from configs, but it's stil an absolute path :/ will look into that again at some point tomorrow

Reply:
okay, I think it works now.